src/Controls/tests/DeviceTests/Elements/Window/WindowTests.Windows.cs

```diff
@@ -7,13 +7,42 @@
 using Microsoft.Maui.Handlers;
 using Microsoft.Maui.Hosting;
 using Microsoft.Maui.Platform;
+using Microsoft.UI.Xaml.Automation.Peers;
+using Microsoft.UI.Xaml.Automation.Provider;
 using Xunit;
 using WPanel = Microsoft.UI.Xaml.Controls.Panel;
 
 namespace Microsoft.Maui.DeviceTests
 {
 	public partial class WindowTests : ControlsHandlerTestBase
 	{
+		[Fact(DisplayName = "Swapping MainPage no Crash")]
+		public async Task SwappingMainPageNoCrash()
+		{
+			SetupBuilder();
+
+			var button = new Button();
+
+			var mainPage = new ContentPage
+			{
+				Content = button
+			};
+
+			var clicked = false;
+
+			button.Clicked += delegate
+			{
+				Application.Current.MainPage = mainPage;
```

Comment:
For the test and I don't think we want to swap out the entire page of the test runner. 

If you look at the `MainPageSwapTests` tests, you'll see how we are standing up an Application/Window that's isolated ot the test

Reply:
Updated the tests.